### PR TITLE
Issue #662: moved ForbidCCommentsInMethodsCheck to log AST

### PR DESCRIPTION
**Pull request by rnveach**
Issue #662 

I rewrote the check to use `visitToken` and log on the AST instead of relying on `getFileContents()`.
Tests show no differences. I will confirm the same with regression.

### CODE REVIEW

File: sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCCommentsInMethodsCheck.java [hunk 1]
```diff
@@ -39,15 +40,21 @@
      */
     public static final String MSG_KEY = "forbid.c.comments.in.the.method.body";
 
-    /**
-     * Set contains C style comments from current file.
-     */
-    private Set<Integer> clangComments;
+    /** AST to use for null checks because {@link Deque} doesn't allow null elements. */
+    private static final DetailAST NULL_AST = new DetailAST();
```
Comment (rnveach) [inline sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCCommentsInMethodsCheck.java]: `Deque` does not allow `null` elements to be placed inside it.
I originally wanted a Boolean to say in method or not, but I need to keep track of the current method AST for checking line numbers.

Making static class to hold the `null` AST seems pointless, so this was my workaround.

**Comment by romani** [comment]
regression report will be good to have, as logic is changed

**Comment by rnveach** [comment]
Regression: http://rveach.no-ip.org/checkstyle/regression/reports/162/

It is showing a ton of differences, so I will need to look into these and see where I went wrong.

**Comment by rnveach** [comment]
http://rveach.no-ip.org/checkstyle/regression/reports/162/openjdk10/index.html#A87
New violations. They are invalid because it is a comment in the middle of a method declaration.
I will fix and add a test case.

**Comment by rnveach** [comment]
> http://rveach.no-ip.org/checkstyle/regression/reports/162/openjdk10/index.html#A87

Fixed and added new test cases. Re-running regression.

**Comment by romani** [comment]
from travis:
`Compilation failure
/home/travis/build/sevntu-checkstyle/sevntu.checkstyle/sevntu-checks/src/test/resources/com/github/sevntu/checkstyle/checks/coding/InputForbidCCommentsInMethodsCheck.java:[84,10] ';' expected
`

**Comment by rnveach** [comment]
New Regression: http://rveach.no-ip.org/checkstyle/regression/reports/163/

**Comment by rnveach** [comment]
@romani I looked through the majority of the violations.
I went through half of the openjdk violations for my first post on regression. I checked through alot of the smaller sets and picked random ones to review.

I don't see anything else so I think we are good. Please review.

**Pull request merged**